### ISSUE TITLE
Deprecation information and notice

### ISSUE THREAD

**Issue by dklimpel** [element-hq/matrix-authentication-service#3464]
Matthew has written: https://matrix.org/blog/2024/10/29/matrix-2.0-is-here/

> In short, Next Gen Auth is transformative, and the initial implementation at [matrix-authentication-service](https://github.com/element-hq/matrix-authentication-service) (MAS) is ready for admins to deploy and use (and in future will be available embedded in Synapse too).

To me this sounds like a depreciation of the MAS. It would be good if there was a schedule and information for users about the future and this was not just described in passing in the blog.

In the end, administrators would now migrate to MAS with a lot of effort and then have to perform the next migration back to Synapse. This would certainly weaken rather than strengthen acceptance.

**Comment by ara4n** [comment]
This is a misunderstanding. The MAS executable is not deprecated(!) and will remain the main way to power OIDC auth in Matrix - particularly in large deployments where MAS might well want to be clustered and deployed independently of the homeserver processes.

However, MAS has also been designed so that it may, *in future*, be embedded into existing server codebases.  To quote https://github.com/element-hq/matrix-authentication-service/blob/main/docs/development/architecture.md#architecture:

> The service is meant to be easily embeddable, with only a dependency to a database. It is also meant to stay lightweight in terms of resource usage and easily scalable horizontally.

In other words, to ease deployment for folks hosting small Matrix deployments and wanting as few moving parts as possible, there's a chance in future that MAS can **optionally** be deployed by being embedded into Synapse.  However, that's not roadmapped currently, and we would *not* recommend it for large Synapse deployments, where you want as much control over what each process is doing as possible, rather than bundling everything into a monolith.
